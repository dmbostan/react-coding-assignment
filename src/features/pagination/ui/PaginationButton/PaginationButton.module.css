.button {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 34px;
  height: 34px;
  margin: 0 8px;
  padding: 0;

  background-color: transparent;
  border: 2px solid var(--color-light);
  border-radius: var(--size-radius);
  cursor: pointer;
}

.button.active {
  background-color: var(--color-light);
}
