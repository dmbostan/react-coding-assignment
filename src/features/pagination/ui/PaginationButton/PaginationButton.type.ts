import {ReactNode} from "react";

export type ButtonTarget = number | 'next' | 'previous' | 'last';

export type PaginationButtonTypes = {
    target: ButtonTarget,
    currentPage: number,
    label: string,
    clickHandler: (page?:ButtonTarget) => void,
    children: ReactNode[] | ReactNode | null
}
