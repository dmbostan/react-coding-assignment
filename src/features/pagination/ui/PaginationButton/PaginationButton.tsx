import React from 'react';
import styles from './PaginationButton.module.css';
import {PaginationButtonTypes} from './PaginationButton.types';

const PaginationButton: React.FC<PaginationButtonTypes> = ({page, currentPage, label, clickHandler, children}) => {

  const activeClass = page === currentPage ? styles.active : '';

  return (
      <button key={page}
              className={`${styles.button} ${activeClass}`}
              aria-label={label}
              onClick={() => clickHandler(page)}>
        {children}
      </button>
  );
};

export default PaginationButton;
