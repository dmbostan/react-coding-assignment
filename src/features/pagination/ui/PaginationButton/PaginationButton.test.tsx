import { render, fireEvent } from '@testing-library/react';
import PaginationButton from './PaginationButton';

describe('PaginationButton', () => {
    it('should render the button with correct label and class when not active', () => {
        const { getByLabelText, getByText } = render(
            <PaginationButton page={1} currentPage={2} label="Page 1" clickHandler={() => {}}>
                Button Text
            </PaginationButton>
        );

        const button = getByLabelText('Page 1');
        const buttonText = getByText('Button Text');

        expect(button).toBeInTheDocument();
        expect(buttonText).toBeInTheDocument();
        console.log()

        expect(button.className).toContain('button');
        expect(button.className).not.toContain('active');
    });

    it('should render the button with correct label and class when active', () => {
        const { getByLabelText, getByText } = render(
            <PaginationButton page={1} currentPage={1} label="Page 1" clickHandler={() => {}}>
                Active Button Text
            </PaginationButton>
        );

        const button = getByLabelText('Page 1');
        const buttonText = getByText('Active Button Text');

        expect(button).toBeInTheDocument();
        expect(buttonText).toBeInTheDocument();
        expect(button.className).toContain('button');
        expect(button.className).toContain('active');
    });

    it('should call the clickHandler when the button is clicked', () => {
        const clickHandlerMock = vi.fn();
        const { getByLabelText } = render(
            <PaginationButton page={1} currentPage={2} label="Page 1" clickHandler={clickHandlerMock}>
                Click Me
            </PaginationButton>
        );

        const button = getByLabelText('Page 1');

        fireEvent.click(button);

        expect(clickHandlerMock).toHaveBeenCalledWith(1);
    });

    it('should match snapshot', () => {

        const { asFragment } = render(
            <PaginationButton page={1} currentPage={2} label="Page 1" clickHandler={() => {}}>
                Click Me
            </PaginationButton>
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
