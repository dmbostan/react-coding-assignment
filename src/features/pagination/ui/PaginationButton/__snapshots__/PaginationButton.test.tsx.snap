// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaginationButton > should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Page 1"
    class="_button_045c1d "
  >
    Click Me
  </button>
</DocumentFragment>
`;
